### ISSUE TITLE
Improve API for setting the formatter in TaurusForms

### ISSUE THREAD

**Issue by mbroseta** [taurus-org/taurus#647]
We have observed a different behavior of the units value when using or a Taurusform or writing the code. When using the Taurusform the units follow the formatter, but when writing the application by the code (below), the units show its long format instead of the tangoFormatter value.

```
from taurus.qt.qtgui.panel import TaurusForm
from taurus.qt.qtgui.application import TaurusApplication
import sys
from taurus.core.tango.util import tangoFormatter
from taurus.qt.qtgui.base import TaurusBaseComponent
TaurusBaseComponent.FORMAT = tangoFormatter
app = TaurusApplication(sys.argv)
tf = TaurusForm()
tf.setModel("sys/tg_test/1/short_scalar_ro")
tf.show()
app.exec_()
```

Br
Manolo.

**Comment by cpascual** [comment]
The reason of the difference is that in your snippet, you are setting the formatter for **all** TaurusBaseComponent classes, which is different to what it is done behind the scenes when you open a taurusform and use the context menu option for setting the formatter.

In order to programmatically emulate what you do on the GUI, you should specifically assign the tango formatter to only the read widget instance of the TaurusValue:

```python
from taurus.qt.qtgui.panel import TaurusForm
from taurus.qt.qtgui.application import TaurusApplication
import sys
from taurus.core.tango.util import tangoFormatter
app = TaurusApplication(sys.argv)
tf = TaurusForm()
tf.setModel("sys/tg_test/1/short_scalar_ro")
tf[0].readWidget().setFormatter(tangoFormatter)  # set formatter at readWidget level
tf.show()
app.exec_()
```

or, if you want to affect all values of all TaurusForms in your application, you could do:

```python
from taurus.qt.qtgui.panel import TaurusForm
from taurus.qt.qtgui.application import TaurusApplication
import sys
from taurus.core.tango.util import tangoFormatter
from taurus.qt.qtgui.panel.taurusvalue import DefaultReadWidgetLabel  # ugly!
DefaultReadWidgetLabel.FORMAT = tangoFormatter
app = TaurusApplication(sys.argv)
tf = TaurusForm()
tf.setModel(["sys/tg_test/1/short_scalar_ro", "sys/tg_test/1/ampli"])
tf.show()
app.exec_()
```

Note that I marked the import of `DefaultReadWidgetLabel`as "ugly". The reason is that you are not suppossed to import from `taurus.qt.qtgui.panel.taurusvalue`, but from  `taurus.qt.qtgui.panel` ... but I just discovered that `DefaultReadWidgetLabel` is not available in `taurus.qt.qtgui.panel`

I think that this is a bug, and I will make a PR correcting it.

Also, I think that while it is possible to do what you want with current API (as shown in the first example), the API could be made simpler so that instead of being forced to set the formatter at the read widget level, you could do it at TaurusValue and/or TaurusForm level as in the following examples (where I omit the boilerplate code)

```python
tf = TaurusForm()
tf.setModel("sys/tg_test/1/short_scalar_ro")
tf[0].setFormatter(tangoFormatter)  # set formatter at TaurusValue level
```

or even:

```python
tf = TaurusForm()
tf.setModel("sys/tg_test/1/short_scalar_ro")
tf.setFormatter(tangoFormatter)  # set formatter at TaurusForm level
```

The enhancement should be easy as well (expect a PR for next release)

**Comment by cpascual** [comment]
In summary, I think that we should:

- Add DefaultReadWidgetLabel to the `__all__` list of `taurus.qt.qtgui.panel.taurusvalue`
- reimplement `setFormatter` for TaurusValue and TaurusForm in a way that is coherent with `onSetFormatter`
- Define `DefaultUnitsWidget.FORMAT` as `{}`  (this would allow @mbroseta's code to work straight away (even if doing such a broad setting is discouraged)

**Comment by cpascual** [comment]
I implemented the changes in #767.
@mbroseta can you check if it works for you?
